bitcoin/src/cli.rs

```diff
@@ -24,27 +27,72 @@ pub struct BitcoinOpts {
     /// is used depending on the detected network.
     #[clap(long)]
     pub electrs_url: Option<String>,
+
+    /// Experimental: Run in light client mode
+    #[clap(long)]
+    pub light: bool,
+
+    /// Wif encoded Bitcoin private key
+    /// If set, we are to use the light client
+    // TODO: load key from file
+    #[clap(long, requires("light"), required_if_eq("light", "true"))]
+    pub private_key: Option<PrivateKey>,
+
+    /// Bitcoin network, only needed for
+    /// configuring the light client
+    #[clap(long, requires("light"), required_if_eq("light", "true"))]
+    pub network: Option<Network>,
 }
 
 impl BitcoinOpts {
     fn new_auth(&self) -> Auth {
-        Auth::UserPass(self.bitcoin_rpc_user.clone(), self.bitcoin_rpc_pass.clone())
+        Auth::UserPass(
+            self.bitcoin_rpc_user.clone().expect("User not set"),
+            self.bitcoin_rpc_pass.clone().expect("Pass not set"),
```

Comment:
To the extent that this only panic if there is a bug, yes. From what I understand this is common practice and since this is at startup it would fail immediately